example/ipsec_offload/odp_ipsec_offload_fwd_db.h

```diff
@@ -0,0 +1,201 @@
+	uint32_t          index;          /**< Next available entry */
+	fwd_db_entry_t   *list;           /**< List of active routes */
+	fwd_db_entry_t    array[MAX_DB];  /**< Entry storage */
+} fwd_db_t;
+
+/** Global pointer to fwd db */
+extern fwd_db_t *fwd_db;
+
+/**
+ * Flow cache table entry
+ */
+typedef struct {
+	void			*next;	/**< Pointer to next flow in list*/
+	uint32_t		l3_src;	/**< Source IP Address*/
+	uint32_t		l3_dst;	/**< Destination IP Address*/
+	ipsec_out_entry_t	out_port; /**< Out interface of matching flow*/
+} flow_entry_t;
+
+/**
+ * Flow cache table bucket
+ */
+typedef struct {
+	odp_spinlock_t		lock;	/**< Bucket lock*/
+	flow_entry_t		*next;	/**< Pointer to first flow entry in bucket*/
+} flow_bucket_t;
+
+/**
+* Pointers to Flow cache tables
+*/
+extern flow_bucket_t *flow_table;
+
+extern flow_bucket_t *ipsec_out_flow_table;
+
+extern flow_bucket_t *ipsec_in_flow_table;
+
+/**
+ * Number of buckets in hash table
+ */
+extern uint32_t bucket_count;
+
+/*
+ * Allocate and Initialize routing table with default Route entries.
+ *
+ */
+void init_routing_table(void);
+
+/*
+ * Searches flow entry in given hash bucket according to given 5-tuple
+ * information
+ *
+ * @param sip           Source IP Address
+ * @param dip           Destination IP Address
+ * @param sport         Source Port Number
+ * @param dport         Destination Port Number
+ * @param proto         IP protocol
+ * @param bucket        Hash Bucket
+ *
+ * @return Matching flow entry
+ */
+static inline flow_entry_t *route_flow_lookup_in_bucket(uint32_t sip,
+							uint32_t dip,
+							void *bucket)
+{
+	flow_entry_t      *flow, *head;
+
+	head = ((flow_bucket_t *)bucket)->next;
+	for (flow = head; flow != NULL; flow = flow->next) {
+		if ((flow->l3_src == sip) && (flow->l3_dst == dip))
+			return flow;
+	}
+	return NULL;
+}
+
+/**
+ * Insert the flow into given hash bucket
+ *
+ * @param flow		Which is to be inserted
+ * @param bucket	Target Hash Bucket
+ *
+ */
+static inline void route_flow_insert_in_bucket(flow_entry_t *flow,
+					       void *bucket)
+{
+	flow_entry_t *temp;
+	flow_bucket_t *bkt = (flow_bucket_t *)bucket;
+
+	if (!flow) {
+		EXAMPLE_ERR("Invalid flow entry passed\n");
+		return;
+	}
+
+	LOCK(&bkt->lock);
+	/*Check that entry already exist or not*/
+	temp = route_flow_lookup_in_bucket(flow->l3_src, flow->l3_dst, bkt);
+	if (temp) {
+		UNLOCK(&bkt->lock);
+		return;
+	}
+
+	if (!bkt->next) {
+		bkt->next = flow;
+	} else {
+		temp = bkt->next;
+		flow->next = temp;
+		bkt->next = flow;
+	}
+	UNLOCK(&bkt->lock);
+}
+
+/** Initialize FWD DB */
+void init_fwd_db(void);
+
+/**
+ * Create a forwarding database entry
+ *
+ * String is of the format "SubNet:Intf:NextHopMAC"
+ *
+ * @param input  Pointer to string describing route
+ *
+ * @param if_names  Array of Name of the interfaces available
+ *
+ * @param if_count  number of interfaces in if_names array
+ *
+ * @param entries number of entries
+ *
+ * @return 0 if successful else -1
```

Comment:
why this file has bunch of empty lines like here? On  other files code is more compact. Let's keep the same code style.